### DISCUSSION TITLE
How to add items to the "dfu.quirk" file and push it successfully?

### DISCUSSION THREAD

**Discussion by SonixLeo** [fwupd/fwupd#5057] [General]
I want to add DFU Quirk items for our company's products, so I need to modify the "dfu.quirk" file.
But when I PUSH the code, I knock over the error message underneath.
May I ask what application should be made to update the DFU.QUIRk file?

------ error message -------
walter@walter:/media/walter/BA8A26CF8A268849/work/dfu/220920-1/fwupd$ git push -u origin main
Username for 'https://github.com': [leo_huang@sonix.com.tw](http://leo_huang@sonix.com.tw/)
Password for 'https://leo_huang@sonix.com.tw@github.com': 
remote: Permission to fwupd/fwupd.git denied to SonixLeo.
fatal: unable to access ['https://github.com/fwupd/fwupd.git/':](http:) The requested URL returned error: 403

**Comment by hughsie** [comment]
You need to fork fwupd, then push the changes to a branch on your fork, then create a pull request on this repo using your branch+fork.

**Reply by SonixLeo** [reply]
Thank you for your guidance
